master/main.go

```diff
@@ -1,5 +1,90 @@
 package main
 
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/websocket"
+	"gopkg.in/mgo.v2"
+)
+
+var connections map[*websocket.Conn]bool
+var session *mgo.Session
+var collUsers *mgo.Collection
+var (
+	upgrader = websocket.Upgrader{
+		ReadBufferSize:  1024,
+		WriteBufferSize: 1024,
+	}
+)
+
+func authentication(loginAndPass []string) bool {
+	usr, err := findByUsername(loginAndPass[0])
+	if err != nil {
+		return false
+	} else if usr.Password == loginAndPass[1] {
```

Comment:
Security vulnerability: this comparison code is vulnerable to a timing attack. To compare authentication and authorization related information use functions from this package: https://golang.org/pkg/crypto/subtle/